components/dada-execute/src/step/apply_op.rs

```diff
@@ -0,0 +1,106 @@
+use dada_ir::{
+    code::{bir, validated::op::Op},
+    error,
+};
+
+use crate::{
+    error::DiagnosticBuilderExt,
+    machine::op::MachineOpExt,
+    machine::{Object, ObjectData, Value},
+};
+
+use super::Stepper;
+
+impl Stepper<'_> {
+    pub(super) fn apply_op(
+        &mut self,
+        expr: bir::Expr,
+        op: Op,
+        lhs: Object,
+        rhs: Object,
+    ) -> eyre::Result<Value> {
+        let op_error = || {
+            let span = self.span_from_bir(expr);
+            Err(error!(
+                span,
+                "cannot apply operator {} to {} and {}",
+                op,
+                self.machine[lhs].kind_str(self.db),
+                self.machine[rhs].kind_str(self.db),
+            )
+            .eyre(self.db))
+        };
+        let div_zero_error = || {
+            let span = self.span_from_bir(expr);
+            Err(error!(span, "divide by zero").eyre(self.db))
+        };
+        let overflow_error = || {
+            let span = self.span_from_bir(expr);
+            Err(error!(span, "overflow").eyre(self.db))
+        };
+        match (&self.machine[lhs], &self.machine[rhs]) {
+            (&ObjectData::Bool(lhs), &ObjectData::Bool(rhs)) => match op {
+                Op::EqualEqual => Ok(self.machine.our_value(lhs == rhs)),
```

Comment:
Why do these new values become "our" instead of "my"?

Reply:
In general, scalar values should be "our" so that they get copied from place to place when they are used. "my" would be for things (like objects) that have identity.

(side note, I've been rethinking some of these names, but that's neither here nor there.)